modules/host/kernel.nix

```diff
@@ -55,19 +67,40 @@
     - also see https://github.com/NixOS/nixpkgs/issues/109280
       for the context >
     */
+    
     configfile = ./ghaf_host_hardened_baseline;
     allowImportFromDerivation = true;
   };
 
-  cfg = config.ghaf.host.kernel_hardening;
+  pkvm_patch = lib.mkIf config.ghaf.hardware.x86_64.common.enable [ {
+    name = "pkvm-patch";
+    patch = ../virtualization/pkvm/0001-pkvm-enable-pkvm-on-intel-x86-6.1-lts.patch;
+    structuredExtraConfig = with lib.kernel; {
+      KVM_INTEL=yes;
+      KSM=no;
+      PKVM_INTEL=yes;
+      PKVM_INTEL_DEBUG=yes;
+      PKVM_GUEST=yes;
+      EARLY_PRINTK_USB_XDBC=yes;
+      RETPOLINE=yes;
+    };
+  } ];
+  
+  kern_cfg = config.ghaf.host.kernel_hardening;
+  hyp_cfg = config.ghaf.host.hypervisor_hardening;
 in
   with lib; {
     options.ghaf.host.kernel_hardening = {
       enable = mkEnableOption "Host kernel hardening";
     };
+    
```

Comment:
At the moment `ghaf.host.kernel_hardening.enable` and  `ghaf.host.hypervisor_hardening.enable` are mutually exclusive. I'm not sure if there's a nixified way to define options that are mutually exclusive - @mikatammi ? 
home-manager options seemed to document that "option x is mutually exclusive with y" and "option y is mutually exclusive with x". That's least we could add to both of these.